packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -625,20 +653,36 @@ export class CallContext {
     });
   }
 
-  public setDeviceManagerUnparentedView(
-    localVideoStream: LocalVideoStreamState,
-    view: VideoStreamRendererViewState | undefined
-  ): void {
+  public setDeviceManagerUnparentedView(args: {
+    localVideoStream: LocalVideoStreamState;
+    view: VideoStreamRendererViewState | undefined;
+    /* @conditional-compile-remove(video-background-effects) */
+    localVideoStreamEffectsAPI: VideoEffectsFeature;
+  }): void {
+    /* @conditional-compile-remove(video-background-effects) */
+    {
+      this._unparentedViewVideoEffectsSubscriber?.unsubscribe();
+      this._unparentedViewVideoEffectsSubscriber = new LocalVideoStreamVideoEffectsSubscriber({
```

Comment:
I can see why we put a subscriber here - cause we only create localStream in StreamUtil function then it's hard to get the stream and subscribe to it in other functions

But this adds complexity to the overall dependency graph, previously call context is only responsible for getting and setting data and all the other logics are in subscriber/client
But now it's like subscriber->context->subscriber->context, so it could be easily a circular ref when someone doesn't understand this well

I guess either finding a way to create stream outside of context or getting stream from internal context after the creation or even creating subscriber in internal context would be nice

Lets keep CallContext only the final stop for data and logic to flow in ;D

Reply:
great point, will look into where else it could go...

Reply:
Moved the subscription logic into internalContext, not super ideal as then the internalcontext kinda has a handle to the regular context but changes
`subscriber->context->subscriber->context`
to
`subscriber->internalcontext->subscriber->context`
which is a little nicer!